include/libtorrent/sha1_hash.hpp

```diff
@@ -262,7 +263,10 @@ namespace libtorrent
 		{
 			return std::string(reinterpret_cast<char const*>(&m_number[0]), size());
 		}
-
+		std::string to_hex() const
```

Comment:
I agree with not adding this as a member function, however,  see article page 3:

> The "best" interface for a class can be found only by balancing many competing concerns, of which the degree of encapsulation is but one.

I do add this function as a member in `jlibtorrent` swig interface :)